src/BugsnagUnity/Delivery.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using BugsnagUnity.Payload;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace BugsnagUnity
+{
+    class Delivery
+    {
+
+        private GameObject _dispatcherObject;
+        private Configuration _configuration;
+        private object _callbackLock { get; } = new object();
+
+        private static List<string> _finishedCacheDeliveries = new List<string>();
+
+        private bool _cacheDeliveryInProcess;
+
+
+        internal Delivery(Configuration configuration)
+        {
+            _configuration = configuration;
+            CreateDispatchBehaviour();
```

Comment:
If this is done in the constructor (like it is in master), I don't see a reason that this needs to be called just before use? This could just be inlined:

```suggestion
            _dispatcherObject = new GameObject("Bugsnag main thread dispatcher");
            _dispatcherObject.AddComponent<MainThreadDispatchBehaviour>();
```